### DISCUSSION TITLE
React-NextJS example

### DISCUSSION THREAD

**Discussion by zdanl** [rainbow-me/rainbowkit#1614] [Q&A]
Hi.
I find Rainbowkit highly functional and appealing. That said, integration into NextJS code bases turns out to be a nightmare, at least for me. Even when the Wagmi and Rainbowkit Context is added correctly in _app.tsx and the <ConnectButton /> containing componen is marked as `"use client"` a server-side mess occurs.

Either it is clearly server-side used dependencies asked to be installed for the client, namely 'fs', or wagmi/rainbowkit functions errror to be called outside the wagmi/rainbow context.

Is there any chance that you could kindly explain this, or provide a solid example on integration with NextJS? [1]

[1] https://www.rainbowkit.com/docs/installation

Thanks,
Dan

**Comment by raffle-x-dev** [comment]
2 months and no answers? Is this project alive?

**Reply by 0xmkh** [reply]
Apologies for the late response

**Comment by 0xmkh** [comment]
@zdanl We have the example listed in the docs [here](https://www.rainbowkit.com/docs/installation)

![image](https://github.com/rainbow-me/rainbowkit/assets/53529533/f590ed52-8e00-4c55-9ac1-58f0fa932af7)

But feel free to checkout our [with-next-app](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next-app) example and [with-next-mint](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next-mint-nft).

What exactly is the issue you're facing ? Can you share an example code ?
